packages/vite/src/node/plugins/html.ts

```diff
@@ -505,12 +554,32 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
         }
 
         const shortEmitName = path.posix.relative(config.root, id)
+        // no use assets plugin because it will emit file
+        let match: RegExpExecArray | null
+        let s: MagicString | undefined
+        while ((match = inlineCSSRE.exec(result))) {
+          s = s || (s = new MagicString(result))
```

Comment:
:eyes: 

Could this be simplified to this? (Not sure if it's the same)

```suggestion
          s ||= new MagicString(result)
```

Reply:
it will compile `s || (s = new MagicString(result));`